package/src/components/ChannelPreview/hooks/useChannelPreviewDisplayName.ts

```diff
@@ -83,33 +85,23 @@ export const useChannelPreviewDisplayName = <
   const { client } = useChatContext<StreamChatGenerics>();
   const { vw } = useViewport();
 
-  const DEFAULT_MAX_CHARACTER_LENGTH = (vw(100) - 16) / 6;
+  const DEFAULT_MAX_CHARACTER_LENGTH = Math.floor((vw(100) - 16) / 6);
 
   const currentUserId = client?.userID;
   const members = channel?.state?.members;
-  const numOfMembers = Object.keys(members || {}).length;
   const channelName = channel?.data?.name;
   const characterLimit = characterLength || DEFAULT_MAX_CHARACTER_LENGTH;
-  const [displayName, setDisplayName] = useState(
-    getChannelPreviewDisplayName({
-      channelName,
-      characterLimit,
-      currentUserId,
-      members,
-    }),
-  );
 
-  useEffect(() => {
-    setDisplayName(
+  const displayName = useMemo(
+    () =>
       getChannelPreviewDisplayName({
         channelName,
         characterLimit,
         currentUserId,
         members,
       }),
-    );
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [channelName, currentUserId, characterLimit, numOfMembers]);
+    [channelName, characterLimit, currentUserId, members],
```

Comment:
We probably aren't sure that `members` will change its reference each time a member is added (for example, it may very well be the fact that it gets mutated rather than copied on write, as we can see in the other hooks).

I would keep both `members` and `numMembers` here

Reply:
(for the time being at least, until we convert `channel` to be fully reactive)

Reply:
Makes sense, added it back. Thanks!